[18.0][MIG] l10n_es_partner: Migration to 18.0

mpascuall opened the pull request:
Module migrated to version 18.0

He tenido que modificar la vista del wizard ya que en Odoo 18 no existe el modelo 'res.config.installer' y lo he añadido al menú Bancos de Facturación.

cc https://github.com/APSL 166331

@miquelalzanillas @lbarry-apsl @javierobcn @peluko00 @ppyczko @BernatObrador please review

Commit:
[REF] *: Refactorización para portar a 6.0 - Paso 1.

- Se han renombrado los módulos para usar la nomenclatura propuesta
  por OpenERP: l10n_es para el módulo base de localización (plan de
  cuentas), l10n_es_* para el resto de módulos.

- Se eliminan los módulos extra_addons/* que deberían moverse a
  los extra-addons genéricos (no son específicos de España).

- Se renombran los __terp__.py por __openerp__.py

Commit:
[FIX] l10n_es_partner: Corregido para ejecutarse correctamente en la v 6.0

Commit:
[REF][REN][FIX] Corregimos errores en los nombres de dependencias de ciertos módulos, por los correspondientes que se modificaron para esta versión 6.0 y en ciertos __init__ adaptamos los imports a los nuevos nombres de los ficheros. Renombrado de los archivos de traducción españoles de es_ES.po a es.po y pequeñas refactorizaciones

Commit:
[FIX+IMP] l10n_es_partner: Varios:

[FIX] Renombrado de los archivos de traducción catalanas de ca_ES.po a ca.po
[IMP] Eliminadas reglas de seguridad en modelos osv.osv_memory
[IMP] Añadido website en la definición del módulo
[IMP] Actualizado listado bancos y cajas españoles con los últimos datos disponibles del Banco de España: http://www.bde.es/webbde/es/secciones/servicio/regis/regent_excel.html
[IMP] Improved the introduction of the VAT (uppercase conversion).
[RFR] Cambio de licencia de GPL3 a AGPL3 en el resto de módulos de localización española, con el consentimiento de todos los autores expresado vía lista de correo del equipo de localización openerp-spain@googlegroups.com

acysos committed:
[FIX+IMP] l10n_es_partner: Varias cosas

* [IMP] Cambio de la denomición del banco 0216. Actualizado por BDE el 22/11/2011
* [IMP] Añadido ING DIRECT a las lista de bancos
* [FIX] Añadido campo code, eliminado en res.bank de la versión 6.1.
* [FIX] Bug #950256

Commit:
[IMP] l10n_es_partner: Improved partner search to search also for comercial name field.

Commit:
[MERGE] with lp:~manu-tiedra/openerp-spain/6.1 [FIX] code field was removed from res.bank so I have added it in the localization as it is required to lookup banks when entering the bank account number. It isn't a normal merge cause of different bzr version in the branches

Commit:
[IMP] l10n_es_partner: Actualizado listado bancos y cajas españoles con los últimos datos disponibles del Banco de España: http://www.bde.es/webbde/es/secciones/servicio/regis/regent_excel.html

acysos committed:
[FIX] l10n_es_partner: añadido Lloyds Bank International

Commit:
Commit inicial de una rama para la versión 7.0. Todos los módulos fueron marcados como no instalables mientras no se compruebe que funcionan o migren, para poder usarlos o probarlos se debería poner el atributo installable de los ficheros __openerp__.py de cada módulo a True

acysos committed:
[FIX] l10n_es_partner: migración a OpenERP 7.0

Commit:
[FIX+IMP] l10n_es_partner: Several things:

[IMP] Refactorización de código
[IMP] Asistente de importación puesto como res.config.installer que se ejecuta al instalar el módulo.
[IMP] Limpieza del archivo XML de bancos.
[FIX] Adaptación de las vistas a la versión 7.0.
[IMP] PEP8 + mejoras en la herencia de vistas

Launchpad automatic translations update.

acysos committed:
[FIX] l10n_es_partner: añadido soporte para IBAN + arreglado con la última versión

Launchpad automatic translations update.

Commit:
[IMP] l10n_es_partner: Adaptación de los import a formato v7/v8
[IMP] l10n_es_partner: Mejorada descripción del módulo.
[IMP] l10n_es_partner: Incluido icono.
[IMP] l10n_es_partner: Añadido BIC de Pastor Serfin.
[FIX] l10n_es_partner: Eliminados espacios antes de comprobar CCC para que dé correcto.
[IMP] l10n_es_partner: Eliminación de la mayoría de warnings que da runbot.

Launchpad automatic translations update.

pedrobaeza committed:
[MIG] l10n_es_partner: Migration to 8.0

rlizana committed:
[FIX+IMP] l10n_es_partner: Varias cosas

* [FIX] Cerrar manejador de fichero y clase en PEP8
* [FIX] Campo code duplicado
* [IMP] Test
* [FIX] Añadida dependecia a base_vat
* [FIX] Aplicar API 8.0 y mas test
* Importar los toponimos

Commit:
[IMP] l10n_es_partner: Generador del archivo de bancos

Commit:
[FIX] l10n_es_partner: Bancos españoles actualizados. Corregido script de generacion de data_banks.xml

pedrobaeza committed:
[IMP] l10n_es_partner: Descripción en README.rst

sergio-teruel committed:
[FIX] l10n_es_partner: Corregir nombre de la tabla en pre-migration.py + Eliminar apóstrofes de los identificadores externos + Forzar cambio de id externo para el banco res_bank_bnpparibasse

pedrobaeza committed:
[IMP] l10n_es_partner: Permitir importación directa desde Internet
Se ha corregido además el script de importación para que ya no cause más problemas

OCA Transbot updated translations from Transifex

rlizana committed:
[FIX] Escapar a acute el fichero de bancos importado desde Internet

[ADD] Mas caracteres a escapar

Commit:
[FIX] l10n_es_partner: prevent imported res.bank data from being removed on update

As the data generated by `l10n.es.partner.import.wizard` is created using
the flag `noupdate=False`, the subsequent update of the module is removing
all the created data. This causes the Bank Accounts (`res.partner.bank`)
to be silently losing their assigned Bank (res.bank).

OCA Transbot updated translations from Transifex

pedrobaeza committed:
[IMP+FIX] l10n_es_partner: Test in Python + use request library + others

carlosdauden committed:
l10n_es_partner: Return commercial name in name_get with configurable pattern

pedrobaeza committed:
[IMP] l10n_es_partner: Searches

* Better search on partner
* Review fixes
* More tests
* Migration script
* Improve search over other terms:

  When searching on the omnibar, you can include other filters and
  the default search includes other different arguments, so the
  condition for adding the commercial name should be expanded.
* Make name_search work

  As Odoo overwrites name_search on res.partner in a non inheritable way,
  we have to include explicitly name_search method and search first for
  the partners with that commercial, and then the rest.

OCA Transbot updated translations from Transifex

pedrobaeza committed:
[MIG] l10n_es_partner: Migration to 10.0

OCA Transbot updated translations from Transifex

pedrobaeza committed:
[MIG] l10n_es_partner: Migration to 11.0

* Updated banks local file.
* Updated import script
* Adapted res.partner.bank logic and views
* Adapted account.journal logic

Change2improve committed:
[11.0][FIX] l10n_es_partner update date in readme.rst

OCA Transbot updated translations from Transifex

carlosdauden committed:
[11.0][IMP] l10n_es_partner: Move validation IBAN to base_bank_from_iban

[UPD] Update l10n_es_partner.pot

oihane committed:
[FIX] l10n_es_partner: missing permission on ir.config_parameter

oihane committed:
[FIX] l10n_es_partner: changing comercial changes display name, fixes #889

pedrobaeza committed:
[MIG] l10n_es_partner: Migration to 12.0

[UPD] Update l10n_es_partner.pot

Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

[UPD] Update l10n_es_partner.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_partner
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_partner/

yajo committed:
[FIX] l10n_es_partner: Hide only HTTP errors

Any problems besides HTTP errors are hidden here, and they shouldn't, because there's no way to debug then.

Commit:
[IMP] l10n_es_partner: Add a context hook in order to hide commercial

Translated using Weblate (Galician)

Currently translated at 100.0% (23 of 23 strings)

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_partner
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_partner/gl/

Translated using Weblate (Spanish)

Currently translated at 100.0% (23 of 23 strings)

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_partner
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_partner/es/

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_partner
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_partner/

Translated using Weblate (Catalan)

Currently translated at 100.0% (23 of 23 strings)

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_partner
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_partner/ca/

[UPD] README.rst

pedrobaeza committed:
[FIX] l10n_es_partner: pass message to logger.exception

Otherwise it doesn't work

[IMP] l10n_es_partner: black, isort

sergio-teruel committed:
[MIG] l10n_es_partner: Migration to v13.0

OCA-git-bot committed:
pre-commit update

ozono committed:
[MIG] l10n_es_partner: Migration to 14.0

ozono committed:
[IMP] l10n_es_partner: black, isort, prettier

victoralmau committed:
[FIX] l10n_es_partner: Change Abanca BIC

danimv5 committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (23 of 23 strings)

Translation: l10n-spain-14.0/l10n-spain-14.0-l10n_es_partner
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-14-0/l10n-spain-14-0-l10n_es_partner/ca/

jabelchi committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (23 of 23 strings)

Translation: l10n-spain-14.0/l10n-spain-14.0-l10n_es_partner
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-14-0/l10n-spain-14-0-l10n_es_partner/ca/

albariera committed:
[FIX] l10_es_partner: fix search

manuelregidor committed:
[MIG]l10n_es_partner: Migration to 15.0

pedrobaeza committed:
[MIG] l10n_es_partner: Migration to 16.0

- Standard procedure.
- Use trigram index for commercial name field.
- Use new feature for declaring the fields to look on when name
  searching
- Remove search override, as it should match only on name search.

TT39999

Commit:
[UPD] Update l10n_es_partner.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-16.0/l10n-spain-16.0-l10n_es_partner
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-16-0/l10n-spain-16-0-l10n_es_partner/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 34.7% (8 of 23 strings)

Translation: l10n-spain-16.0/l10n-spain-16.0-l10n_es_partner
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-16-0/l10n-spain-16-0-l10n_es_partner/sl/

ljsalvatierra-factorlibre committed:
[FIX] l10n_es_partner: _rec_names_search

OCA-git-bot committed:
l10n_es_partner 16.0.1.0.1

OCA-git-bot committed:
[UPD] README.rst

pedrobaeza committed:
[FIX] l10n_es_partner: Bank data download URL + local file fallback

It seems that BDE is mangling somehow the automated download of the
file detecting some patterns. We change the URL for trying to maximize
the possibilities of being downloaded, but also provide a local file
as a fallback.

Commit:
[UPD] Update l10n_es_partner.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-16.0/l10n-spain-16.0-l10n_es_partner
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-16-0/l10n-spain-16-0-l10n_es_partner/

pedrobaeza committed:
[FIX] l10n_es_partner: Wizard UI colspan

The elements are not shown correctly due to the styling changes in v16.

OCA-git-bot committed:
[BOT] post-merge updates

Javiobcn committed:
[IMP] l10n_es_partner: pre-commit auto fixes

Javiobcn committed:
[MIG] l10n_es_partner: Migration to 17.0

Commit:
[UPD] Update l10n_es_partner.pot

OCA-git-bot committed:
[BOT] post-merge updates

pedrobaeza committed:
[FIX] l10n_es_partner: ruff linter

etobella committed:
[FIX] l10n_es_partner: Show address

OCA-git-bot committed:
[BOT] post-merge updates

mpascuall commented:
@pedrobaeza Buenas! En esta migración fallan los tests debido a que da un error descargando la información desde internet, pero en local me pasan correctamente. Debo hacer algún cambio?

juancarlosonate-tecnativa commented:
Justo acabo de realizar una migración con un problema similar relacionado con `res_config_installer`, revisa https://github.com/OCA/l10n-spain/pull/4027, es la alternativa más óptima.

mpascuall commented:
> Justo acabo de realizar una migración con un problema similar relacionado con `res_config_installer`, revisa #4027, es la alternativa más óptima.

Buenas @juancarlosonate-tecnativa! Muchas gracias por tu comentario, básicamente he hecho lo mismo, crear una nueva vista con el wizard manteniendo la misma funcionalidad. Lo único es que hay un problema con los tests que fallan por el warning de la conexión a internet.

EmilioPascual commented:
> > Justo acabo de realizar una migración con un problema similar relacionado con `res_config_installer`, revisa #4027, es la alternativa más óptima.
> 
> Buenas @juancarlosonate-tecnativa! Muchas gracias por tu comentario, básicamente he hecho lo mismo, crear una nueva vista con el wizard manteniendo la misma funcionalidad. Lo único es que hay un problema con los tests que fallan por el warning de la conexión a internet.

No te fijes en el wizard, fijate en el test y como resuelve el problema https://github.com/OCA/l10n-spain/pull/4027/files#diff-995bf288bfd4b78d663bac240b335fd72353933ba075274853f3e608bade9f80R17

mpascuall commented:
> > > Justo acabo de realizar una migración con un problema similar relacionado con `res_config_installer`, revisa #4027, es la alternativa más óptima.
> > 
> > 
> > Buenas @juancarlosonate-tecnativa! Muchas gracias por tu comentario, básicamente he hecho lo mismo, crear una nueva vista con el wizard manteniendo la misma funcionalidad. Lo único es que hay un problema con los tests que fallan por el warning de la conexión a internet.
> 
> No te fijes en el wizard, fijate en el test y como resuelve el problema https://github.com/OCA/l10n-spain/pull/4027/files#diff-995bf288bfd4b78d663bac240b335fd72353933ba075274853f3e608bade9f80R17

Perfecto, muchas gracias!

pedrobaeza commented:
El problema que se está tratando en #4029 afecta aquí también

/ocabot migration l10n_es_partner

mpascuall commented:
> El problema que se está tratando en #4029 afecta aquí también
> 
> /ocabot migration l10n_es_partner

Gracias por tu comentario  @pedrobaeza! ¿Debería añadir un commit específico para este FIX? Ahora mismo he agregado los cambios en el commit de migración, pero si ves que es mejor en un commit específico, lo cambiaré.

pedrobaeza commented:
El tema es que ese parche no vale, por lo que hay que arreglarlo tanto en 17 como en 18.

pedrobaeza committed:
[FIX] l10n_es_partner: Inject _rec_names_search properly

Previous patch was overriding _auto_init method for injecting the
comercial field in the list of fields to search for (_rec_names_search),
but this method is only called on module installation/update, so if
you restart the Odoo server after that, you lose the injection.

Thus, let's inject it if not present when calling name_search, and
this way we always have it available.

TT55241

pedrobaeza commented:
Haz por favor cherry-pick de #4058 antes de los commit de migración

mpascuall committed:
[MIG] l10n_es_partner: Migration to 18.0

mpascuall commented:
> Haz por favor cherry-pick de #4058 antes de los commit de migración

Buenas @pedrobaeza, gracias por el comentario. He hecho el cherry-pick antes del commit de migración pero ahora el test da un ConnectionTimeour, a qué se puede deber?

pedrobaeza commented:
El problema es ajeno a vosotros, no te preocupes. Luego reviso.

juancarlosonate-tecnativa commented:
Hay que añadir https://github.com/OCA/l10n-spain/pull/4131 , el ConnectTimeoutError sigue sucediendo?

pedrobaeza commented on l10n_es_partner/wizard/l10n_es_partner_wizard.xml:
```diff
@@ -55,10 +60,11 @@
-        <field name="action_id" ref="action_l10n_es_partner_import" />
-        <field name="sequence">20</field>
-    </record>
+    <menuitem
```
No se debe crear un menú, si no seguir haciéndolo como un asistente de configuración. Sería crear un registro `ir.actions.todo` como éste:

https://github.com/OCA/l10n-spain/blob/0c75c946c954f5b567ee325fb10577fe84ecc7e6/l10n_es_toponyms/wizard/l10n_es_toponyms_wizard.xml#L51

mpascuall replied:
Hecho, gracias!

pedrobaeza reviewed (approved):
/ocabot merge nobump
